augur/dates/__init__.py

```diff
@@ -121,6 +121,12 @@ def get_numerical_date_from_value(value, fmt=None, min_max_year=None):
         except InvalidDate as error:
             raise AugurError(str(error)) from error
         return [treetime.utils.numeric_date(d) for d in ambig_date]
+    if value.startswith('[') and value.endswith(']') and len(value[1:-1].split(':'))==2:
+        # Treetime ambiguous date format, e.g. [2019.5:2020.5]
```

Comment:
I'm curious how you get the numerical dates in metadata. Is this a standard format outside of TreeTime, or do you run ISO dates through something like `treetime.utils.numeric_date` to get the float representation?

Asking because I think a good next step after this gets merged is to generalize for any `[A:B]` date pair where `A` and `B` can be anything that's currently supported as an exact scalar date (e.g. `[2023-07-31:2023-08-06]`). I did some [prototyping](https://github.com/nextstrain/augur/blob/95476eba1df2c330f0bc5beece8621c403697277/augur/dates/augur_date.py#L22-L23) of this a while back but never got back to creating a proper PR for it.